app/components/landing/projectSection/ResponsiveSearchInput.tsx

```diff
@@ -0,0 +1,117 @@
+'use client';
+
+import { ChangeEvent, useState } from 'react';
+
+import FilterIcon from '@mui/icons-material/FilterAlt';
+import SearchIcon from '@mui/icons-material/Search';
+import Box from '@mui/material/Box';
+import FormControl from '@mui/material/FormControl';
+import IconButton from '@mui/material/IconButton';
+import InputAdornment from '@mui/material/InputAdornment';
+import type { Theme } from '@mui/material/styles';
+import SwipeableDrawer from '@mui/material/SwipeableDrawer';
+import TextField from '@mui/material/TextField';
+import Typography from '@mui/material/Typography';
+import useMediaQuery from '@mui/material/useMediaQuery';
+
+import { closeIconButtonStyle } from '@/components/common/CustomDialog';
+import SecondaryIconButton from '@/components/common/SecondaryIconButton';
+import CloseIcon from '@/components/icons/CloseIcon';
+import ApplyFilterButton, { APPLY_BUTTON } from '@/components/newsPage/newsFilter/ApplyFilterButton';
+import * as m from '@/src/paraglide/messages.js';
+import theme from '@/styles/theme';
+
+type SearchInputProps = { onChange: (event: ChangeEvent<HTMLInputElement>) => void };
+
+export const ResponsiveSearchInput = (props: SearchInputProps) => {
+  const isWideScreen = useMediaQuery((theme: Theme) => theme.breakpoints.up('md'));
+
+  return isWideScreen ? <FullSearchInput {...props} /> : <MobileSearchInput {...props} />;
+};
+
+const FullSearchInput = ({ onChange }: SearchInputProps) => (
+  <FormControl variant="standard" sx={{ width: '100%' }}>
+    <TextField
+      onChange={onChange}
+      variant="outlined"
+      fullWidth
+      sx={{
+        '& .MuiOutlinedInput-root': {
+          '& fieldset': {
+            borderColor: 'white',
+          },
+          '&:hover fieldset': {
+            borderColor: 'white',
+          },
+          '&.Mui-focused fieldset': {
+            borderColor: 'white',
+          },
+          '& input': {
+            color: 'white',
+          },
+        },
+      }}
+      InputProps={{
+        startAdornment: (
+          <InputAdornment position="start">
+            <SearchIcon sx={{ color: 'white' }} />
+          </InputAdornment>
+        ),
+      }}
+      inputProps={{
+        'aria-label': 'Search field',
+      }}
+    />
+  </FormControl>
+);
+
+const MobileSearchInput = ({ onChange }: SearchInputProps) => {
+  const [open, setOpen] = useState(false);
+
+  const openDrawer = () => setOpen(true);
+  const closeDrawer = () => setOpen(false);
+
+  return (
+    <>
+      <SecondaryIconButton
```

Comment:
make sure the filter button is aligned well on mobile
<img width="313" alt="image" src="https://github.com/user-attachments/assets/849cbee5-c5ea-420f-b877-b95716a50ab3">